src/System.Private.ServiceModel/src/Extensions/X509Certificate2Extensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.ServiceModel
+{
+    internal static class X509Certificate2Extensions
+    {
+        // dotnet/wcf#1574 - This is a workaround for dotnet/corefx#12214
+        // The copy constructor was removed in .NET Core, this is the workaround to allow us to make copies of certificates
+
+        internal static X509Certificate2 CloneCertificateInternal(this X509Certificate2 certificateToClone)
+        {
+#if TARGETS_WINDOWS
+            return new X509Certificate2(certificateToClone.Handle);
+#else
+            X509Certificate2Collection collection = new X509Certificate2Collection(certificateToClone);
```

Comment:
> Just to be sure -- the collection does not dispose the certs it owns when it gets GC'ed, right?

Correct.  The collection is just a pre-generics typed collection; no finalizer, no dispose.  Just a longer-named List\<X509Certificate2\> minus some benefits of list and plus the Find method.